components/prism-factor.js

```diff
@@ -0,0 +1,371 @@
+(function (Prism) {
+
+	var comment_inside = {
+		function: {
+			pattern: /(\b)(?:TODO|FIX|FIXME|NOTE|BUG|XX+|HACK|\?\?+|!!+)(?=\b|:)/,
+			lookbehind: true
+		}
+	};
+	var string_inside = {
+		number: /(?:\\[^\s']|%\w)/
+	};
+
+	var factor = {
+
+		'comment': [
+			{
+				// ! single-line exclamation point comments with whitespace after/around the !
+				pattern: /(^|\s)(?:! .*|!$)/,
+				lookbehind: true,
+				inside: comment_inside
+			},
+
+			/* from basis/multiline: */
+			{
+				// /* comment */, /* comment*/
+				pattern: /(^|\s)\/\*\s(?:.|\s)*?\*\/(?=\s|$)/,
```

Comment:
The `(?:.|\s)` is no good. `.` is defined as _all characters except for line breaks_ and `\s` as _all white space characters_ (including line breaks). So `(?:.|\s)` is the same as `[\s\S]` but with the additional problem that `(?:.|\s)` might cause exponential backtracking if the regex engine doesn't optimize.